### DISCUSSION TITLE
Github Pages Development Branch

### DISCUSSION THREAD

**Discussion by joseph-eccles** [community/community#28077] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi there,

I am new to using Github Pages and was hoping someone in the community could possibly give me some advice.

I would like to be able to have a repository that contains different branches such as master, develop staging etc. Each branch would have its own Github Pages site. Example, when push to develop branch a Github Pages Develop site would then be updated.

From what I can see Github Pages only allows you to have one Github Pages site per repository and not per branch.

Has anyone else had such a requirement and if so what solution did they come up with?

Any help would be very much appreciated.

Thanks!

**Comment by ng-druid** [comment, marked as answer]
We have the same issue and have resorted to separate repos with environment suffix. However, I recall reading that github has this capability in beta testing. Looking forward to a general release of that.

**Comment by airtower-luna** [comment]
> From what I can see Github Pages only allows you to have one Github Pages site per repository and not per branch.

That is correct. If you want to have the page deployed you'll need to create a second repository. Note that a local repository can have as many remotes as you want, so you can just push the development branch to another repository. See: [Git Basics - Working with Remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes)

With the [new custom workflow approach](https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/) for pages you could also run the same build on your dev branch and download the artifact to inspect locally instead of deploying. :slightly_smiling_face:

**Comment by joseph-eccles** [comment]
Thanks for the quick response everyone, really appreciate it. @ng-druid thanks for tip about the capability in beta testing. I'll keep an eye on this, do you know if they have given any kind of idea when this could possibly released? Would you be able to explain a little more/add a link about the environment suffix and how it works for your setup?

@airtower-luna thanks for the reply. We're currently trying to have just the one repository for the entire project and all assets (Monorepo) so that's where I am hitting a bit of a blocker as we ideally only have one repository. I was looking at the possibility of having separate repositories and using a workflow to copy a branch to another repository when for example a push is done to the develop branch a develop repository is updated, but with the one repository restriction this can't be done. As you mention maybe local testing of a branch is the only way of doing this for now.

**Reply by ng-druid** [reply]
@joseph-eccles 

Our org has two separate environments dev and prod. Yielding two repos for each pages site.

* [Dev Objects Repo](https://github.com/rollthecloudinc/ipe-objects/tree/dev)
* [Prod Objects Repo](https://github.com/rollthecloudinc/ipe-objects-prod)

The name of "site" is "ipe-objects" therefore the environment suffix for **prod** is "ipe-objects-**prod**".

* [Dev Objects Website](https://rollthecloudinc.github.io/ipe-objects/panelpage/c720b59b-a26f-4919-bab1-219c358b5b68.json)
* [Prod Objects Website](https://rollthecloudinc.github.io/ipe-objects-prod/panelpage/c720b59b-a26f-4919-bab1-219c358b5b68.json)

Also, might I suggest a [build repo](https://github.com/rollthecloudinc/ipe-build) instead of hosting the site from the same repo as the [source code](https://github.com/rollthecloudinc/ipe). This provides much more flexibility. Our orgs setup is anything merged into dev or master kicks off a github action workflow that builds the project and updates the associated build repo. The details of which can be found in our framework core [workflow file](https://github.com/rollthecloudinc/druid/blob/master/.github/workflows/druid-workflow.yml). Each site created contains a workflow file for [each environment](https://github.com/rollthecloudinc/ipe/tree/master/.github/workflows) that uses the previously mentioned reusable workflow.

**Reply by joseph-eccles** [reply]
@ng-druid 

This is a great help thanks so much. This is the way we will approach it for our project. Really appreciate you taking the time to help me on this.
